tests/run.rs

```diff
@@ -45,6 +50,50 @@ fn simple() -> R<()> {
     Ok(())
 }
 
+mod yaml_parse_errors {
+    use super::*;
+
+    #[test]
+    fn wrong_types() -> R<()> {
+        let script = TempFile::write_temp_script("")?;
+        let result = test_run_check_protocols(
+            &script,
+            r##"
+                |protocol: 42
+            "##,
+        );
+        assert_eq!(
+            format!("{}", result.unwrap_err()),
+            format!(
+                "error parsing {}.protocols.yaml: expected: array, got: Integer(42)",
```

Comment:
if i'm reading it correctly, this is not a YAML mistake, it's a misuse of the library. it would be nice if the error message could clarify between the two, so that one tells me (essentially) "you made a YAML mistake" and the other tells me "you misread the `check-protocols` docs"

Reply:
Yeah, that's correct. I changed the error messages now, Are they better like that?

Reply:
yup 👍